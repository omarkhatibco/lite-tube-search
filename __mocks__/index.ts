export * from './response';
export * from './singleResponse';
