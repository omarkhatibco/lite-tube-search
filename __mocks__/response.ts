import { SearchListResponse } from 'types';

export const response: SearchListResponse = {
  kind: 'youtube#searchListResponse',
  etag: 'DKEWsalaJSZw7kWUDnKCN7RwD9U',
  nextPageToken: 'CAkQAA',
  regionCode: 'DE',
  pageInfo: {
    totalResults: 40057,
    resultsPerPage: 9,
  },
  items: [
    {
      kind: 'youtube#searchResult',
      etag: 'mmOrKy-oJXk2DxkN-uWVYBU2xuI',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCXuqSBlHAE6Xw-yeJA0Tunw',
      },
      snippet: {
        publishedAt: '2008-11-25T00:46:52Z',
        channelId: 'UCXuqSBlHAE6Xw-yeJA0Tunw',
        title: 'Linus Tech Tips',
        description:
          'Looking for a Tech YouTuber? Linus Tech Tips is a passionate team of "professionally curious" experts in consumer technology ...',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLSZnNA3eg_fDwUgsCxpqMnVWu3UGo-Rln4621ncIQ=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLSZnNA3eg_fDwUgsCxpqMnVWu3UGo-Rln4621ncIQ=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLSZnNA3eg_fDwUgsCxpqMnVWu3UGo-Rln4621ncIQ=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'Linus Tech Tips',
        liveBroadcastContent: 'upcoming',
        publishTime: '2008-11-25T00:46:52Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: 'WDR6z_iTPdvZzap0zu0PAt7eHc0',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCd294ZJCNIthgHz9I2odC8A',
      },
      snippet: {
        publishedAt: '2016-05-23T19:06:22Z',
        channelId: 'UCd294ZJCNIthgHz9I2odC8A',
        title: 'M1M Tech Channel',
        description:
          'Mein Name ist Marc und hier auf meinem Kanal dreht sich alles um Audio, Lifestyle und Technik Für alle Anfragen Email: ...',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLQT6keUUNAzkz4fDHFOOaY5EgKUTa3yX7E7u9q7mA=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLQT6keUUNAzkz4fDHFOOaY5EgKUTa3yX7E7u9q7mA=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLQT6keUUNAzkz4fDHFOOaY5EgKUTa3yX7E7u9q7mA=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'M1M Tech Channel',
        liveBroadcastContent: 'none',
        publishTime: '2016-05-23T19:06:22Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: 'Opbux-VUECF4P9fkPCzg25-36ig',
      id: {
        kind: 'youtube#channel',
        channelId: 'UC4Tklxku1yPcRIH0VVCKoeA',
      },
      snippet: {
        publishedAt: '2015-06-08T13:31:37Z',
        channelId: 'UC4Tklxku1yPcRIH0VVCKoeA',
        title: 'Quantum Tech HD',
        description:
          'Hello everyone! Welcome to our channel! Here, you can find the most incredible gadgets and inventions, tips and hacks for your ...',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLRjVM0ZkjUBISPAwoeeEcnr1A7tIQptJLiJz_JKyw=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLRjVM0ZkjUBISPAwoeeEcnr1A7tIQptJLiJz_JKyw=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLRjVM0ZkjUBISPAwoeeEcnr1A7tIQptJLiJz_JKyw=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'Quantum Tech HD',
        liveBroadcastContent: 'none',
        publishTime: '2015-06-08T13:31:37Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: '10RflePJ1Kt9yV1juuEsU-uDa1I',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCi26H8N1bzj-b4Nipzdli0w',
      },
      snippet: {
        publishedAt: '2021-03-08T22:56:01Z',
        channelId: 'UCi26H8N1bzj-b4Nipzdli0w',
        title: 'NerdsHeaven Tech',
        description:
          'Reviews, Gadgets, Gaming und der tägliche Nerd-Wahnsinn. Dies ist lediglich ein Zweitkanal. Zum Hauptkanal geht es hier ...',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLR4xAio--UCEqittZwMmYlpzMWms36MTrrOP6qx=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLR4xAio--UCEqittZwMmYlpzMWms36MTrrOP6qx=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLR4xAio--UCEqittZwMmYlpzMWms36MTrrOP6qx=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'NerdsHeaven Tech',
        liveBroadcastContent: 'upcoming',
        publishTime: '2021-03-08T22:56:01Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: 'D6G3xifW-EQh8he12KPloL_iTN0',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCLr4hMhk_2KE0GUBSBrspGA',
      },
      snippet: {
        publishedAt: '2013-10-01T00:23:46Z',
        channelId: 'UCLr4hMhk_2KE0GUBSBrspGA',
        title: 'Kuba Klawiter',
        description:
          'Ten typ od Tech Weeka, testów sprzętu i gadania o technologii.',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLRm7hw2BzMpiOrELuBT4CUC1j1bIqUj5C38zEyk=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLRm7hw2BzMpiOrELuBT4CUC1j1bIqUj5C38zEyk=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLRm7hw2BzMpiOrELuBT4CUC1j1bIqUj5C38zEyk=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'Kuba Klawiter',
        liveBroadcastContent: 'upcoming',
        publishTime: '2013-10-01T00:23:46Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: 'MYiDk9vhOoIvGXvRHeIdoo1Ahz0',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCmoWIa_TL2s5Q14vDbZVwkQ',
      },
      snippet: {
        publishedAt: '2021-08-27T09:26:41Z',
        channelId: 'UCmoWIa_TL2s5Q14vDbZVwkQ',
        title: 'The Tech سعید جولا',
        description:
          'با چی حال میکنی؟ آنباکس گرافیک پردازنده گیم بازی کامپیوتر ماین ترید فارکس بورس بررسی قطعتات اورکلاک کولر شیبا چیا ...',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/NYYx4y7O_1_XJXVN5xTw954nMQ03ZDHD0pf4zwBqC8h6cLkytQMRzWhqKZkIbq_9AlgKX_OtScA=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/NYYx4y7O_1_XJXVN5xTw954nMQ03ZDHD0pf4zwBqC8h6cLkytQMRzWhqKZkIbq_9AlgKX_OtScA=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/NYYx4y7O_1_XJXVN5xTw954nMQ03ZDHD0pf4zwBqC8h6cLkytQMRzWhqKZkIbq_9AlgKX_OtScA=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'The Tech سعید جولا',
        liveBroadcastContent: 'none',
        publishTime: '2021-08-27T09:26:41Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: '0BS5a9NPoGYi00_Qos6aJOuo9pM',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCDeRsDqcB8zdkzErxrdHCOw',
      },
      snippet: {
        publishedAt: '2014-06-25T10:26:34Z',
        channelId: 'UCDeRsDqcB8zdkzErxrdHCOw',
        title: 'TheVR Tech',
        description:
          "TheVR Tech - Quality Videos from Hungary. --- Important notice: We'll only reach you out from our domain (@thevr.hu) if you ...",
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/6L_0neE7CrVV6uwRw0MceohDc0t7fEizeEM_XykBc-R304GE_1Nag9Ljk8Q_h0JgIEcCBgnz_g=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/6L_0neE7CrVV6uwRw0MceohDc0t7fEizeEM_XykBc-R304GE_1Nag9Ljk8Q_h0JgIEcCBgnz_g=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/6L_0neE7CrVV6uwRw0MceohDc0t7fEizeEM_XykBc-R304GE_1Nag9Ljk8Q_h0JgIEcCBgnz_g=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'TheVR Tech',
        liveBroadcastContent: 'none',
        publishTime: '2014-06-25T10:26:34Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: 'irT2QgFGp8kfrTH9-4O9blHWLbU',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCS4wQaIQ_rqrc8LpbvdbrEA',
      },
      snippet: {
        publishedAt: '2020-02-27T10:59:15Z',
        channelId: 'UCS4wQaIQ_rqrc8LpbvdbrEA',
        title: 'OSSRA TECH',
        description:
          'OSSRA TECH unboxing, review, experience, and comparison all in your channel :)',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/49aB62_XxS9SPYWThH_1gyOqJyrWEZe_YEooxshSJLLGexvF1UVdjkIzpwpv5ry81R0-QUY8GtE=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/49aB62_XxS9SPYWThH_1gyOqJyrWEZe_YEooxshSJLLGexvF1UVdjkIzpwpv5ry81R0-QUY8GtE=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/49aB62_XxS9SPYWThH_1gyOqJyrWEZe_YEooxshSJLLGexvF1UVdjkIzpwpv5ry81R0-QUY8GtE=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'OSSRA TECH',
        liveBroadcastContent: 'none',
        publishTime: '2020-02-27T10:59:15Z',
      },
    },
    {
      kind: 'youtube#searchResult',
      etag: 'qI072HGARJ48z6mGp_XaDLQphzw',
      id: {
        kind: 'youtube#channel',
        channelId: 'UCXUJJNoP1QupwsYIWFXmsZg',
      },
      snippet: {
        publishedAt: '2014-09-26T18:24:03Z',
        channelId: 'UCXUJJNoP1QupwsYIWFXmsZg',
        title: 'Tech Burner',
        description:
          'Tech Videos that count! Join Us :) Brands and Business : business@techburner.in This Channel is owned by Burner Media owned ...',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLSUorX6q2Eke1cY_zArinN0Ln-S3ajX2DfXTvWpzw=s88-c-k-c0xffffffff-no-rj-mo',
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLSUorX6q2Eke1cY_zArinN0Ln-S3ajX2DfXTvWpzw=s240-c-k-c0xffffffff-no-rj-mo',
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLSUorX6q2Eke1cY_zArinN0Ln-S3ajX2DfXTvWpzw=s800-c-k-c0xffffffff-no-rj-mo',
          },
        },
        channelTitle: 'Tech Burner',
        liveBroadcastContent: 'upcoming',
        publishTime: '2014-09-26T18:24:03Z',
      },
    },
  ],
};
