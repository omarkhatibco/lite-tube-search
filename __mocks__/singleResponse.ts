import { SearchListResponse } from 'types';

export const singleResponse: SearchListResponse = {
  kind: 'youtube#channelListResponse',
  etag: '1DivoBJVb-IXGMW4aau5eEV8LWY',
  pageInfo: {
    totalResults: 1,
    resultsPerPage: 5,
  },
  items: [
    {
      kind: 'youtube#channel',
      etag: 'bNgdY9b_pJ0xeb1xgsqgpZMzBx8',
      id: 'UCV0qA-eDDICsRR9rPcnG7tw',
      snippet: {
        title: 'Joma Tech',
        description:
          'I talk about life in Silicon Valley and at big tech companies.\n\nFollow me on:\nInstagram: http://instagram.com/jomaoppa\nTwitter: https://twitter.com/jomaoppa\n\nBusiness inquiries: business@joma.io\n',
        customUrl: 'jomaoppa',
        publishedAt: '2016-08-31T22:22:07Z',
        thumbnails: {
          default: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLTFLnUPOScnD78_Lz7YCbp0PXSpPgBTI4HA0pA8=s88-c-k-c0x00ffffff-no-rj',
            width: 88,
            height: 88,
          },
          medium: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLTFLnUPOScnD78_Lz7YCbp0PXSpPgBTI4HA0pA8=s240-c-k-c0x00ffffff-no-rj',
            width: 240,
            height: 240,
          },
          high: {
            url: 'https://yt3.ggpht.com/ytc/AKedOLTFLnUPOScnD78_Lz7YCbp0PXSpPgBTI4HA0pA8=s800-c-k-c0x00ffffff-no-rj',
            width: 800,
            height: 800,
          },
        },
        localized: {
          title: 'Joma Tech',
          description:
            'I talk about life in Silicon Valley and at big tech companies.\n\nFollow me on:\nInstagram: http://instagram.com/jomaoppa\nTwitter: https://twitter.com/jomaoppa\n\nBusiness inquiries: business@joma.io\n',
        },
        country: 'US',
      },
      contentDetails: {
        relatedPlaylists: {
          likes: '',
          uploads: 'UUV0qA-eDDICsRR9rPcnG7tw',
        },
      },
      statistics: {
        viewCount: '133488048',
        subscriberCount: '1810000',
        hiddenSubscriberCount: false,
        videoCount: '100',
      },
    },
  ],
};
