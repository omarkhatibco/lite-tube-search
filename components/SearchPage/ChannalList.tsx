import { Button, Center, SimpleGrid, VStack } from '@chakra-ui/react';
import {
  ChannelListItem,
  ChannelListItemSkeleton,
} from 'components/SearchPage';
import { useChannalSearch } from 'hooks/useChannelSearch';

export const ChannalList: React.FC = () => {
  const { items, isLoading, hasMore, loadMore, perPage } = useChannalSearch();

  return (
    <VStack w={'full'} align='stretch' spacing={8}>
      <SimpleGrid spacing={12} columns={{ base: 1, md: 3 }}>
        {items.map((item) => (
          <ChannelListItem key={item.id?.channelId} {...item} />
        ))}
        {isLoading &&
          Array(perPage)
            .fill('')
            .map(({}, index) => <ChannelListItemSkeleton key={index * -1} />)}
      </SimpleGrid>
      {hasMore && (
        <Center>
          <Button
            isLoading={isLoading}
            isDisabled={isLoading}
            onClick={loadMore}
          >
            Load More
          </Button>
        </Center>
      )}
    </VStack>
  );
};
