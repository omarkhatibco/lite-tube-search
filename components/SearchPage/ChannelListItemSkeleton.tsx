import {
  Center,
  Skeleton,
  SkeletonCircle,
  SkeletonText,
  VStack,
} from '@chakra-ui/react';

export const ChannelListItemSkeleton: React.FC = () => {
  return (
    <VStack as='article' spacing={4} align={'stretch'} w='full'>
      <Center>
        <SkeletonCircle size='36' />
      </Center>
      <VStack align={'stretch'} h='full'>
        <Skeleton w='74%' h='9' />
        <SkeletonText noOfLines={3} skeletonHeight='4' spacing='2' />
      </VStack>
      <Center>
        <Skeleton w='50%' h='10' />
      </Center>
    </VStack>
  );
};
