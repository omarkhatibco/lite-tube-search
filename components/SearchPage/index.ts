export * from './ChannalList';
export * from './ChannelListItem';
export * from './ChannelListItemSkeleton';
