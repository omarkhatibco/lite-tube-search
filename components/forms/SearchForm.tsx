import {
  Button,
  FormControl,
  FormErrorMessage,
  HStack,
  Icon,
  IconButton,
  Input,
  InputGroup,
  InputLeftElement,
  InputRightElement,
  VStack,
} from '@chakra-ui/react';
import { useFormik } from 'formik';
import { MdClear, MdSearch } from 'react-icons/md';
import * as Yup from 'yup';

const validationSchema = Yup.object().shape({
  search: Yup.string()
    .min(3, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
});

export const SearchForm: React.FC = () => {
  const formik = useFormik({
    initialValues: {
      search: '',
    },
    validationSchema,
    onSubmit: (values) => {
      alert(JSON.stringify(values, null, 2));
    },
    onReset: () => {},
  });

  console.log(formik);

  return (
    <form onSubmit={formik.handleSubmit}>
      <HStack align={'flex-start'}>
        <FormControl isInvalid={formik.errors.search && formik.touched.search}>
          <VStack w='full' align={'stretch'}>
            <InputGroup size='md'>
              <InputLeftElement
                pointerEvents='none'
                children={<Icon as={MdSearch} color='red.300' />}
              />
              <Input
                pr='4.5rem'
                name='search'
                type={'search'}
                onChange={formik.handleChange}
                value={formik.values.search}
                isInvalid={formik.errors.search && formik.touched.search}
                placeholder='Search Channels...'
                sx={{
                  '::-webkit-search-decoration,::-webkit-search-cancel-button,::-webkit-search-results-button,::-webkit-search-results-decoration':
                    {
                      display: 'none',
                    },
                }}
              />

              <InputRightElement width='4.5rem'>
                <IconButton
                  icon={<Icon as={MdClear} />}
                  size='sm'
                  variant={'ghost'}
                  onClick={() => formik.resetForm()}
                  rounded='full'
                />
              </InputRightElement>
            </InputGroup>
            {formik.errors.search && formik.touched.search && (
              <FormErrorMessage>{formik.errors.search}</FormErrorMessage>
            )}
          </VStack>
        </FormControl>

        <Button type='submit'>Search</Button>
      </HStack>
    </form>
  );
};
