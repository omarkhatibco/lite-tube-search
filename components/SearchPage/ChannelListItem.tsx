import {
  AspectRatio,
  Center,
  Heading,
  Image,
  Text,
  VStack,
} from '@chakra-ui/react';
import { NextChakraButton } from 'components/wrappers';
import { ChannelItem } from 'types';

export const ChannelListItem: React.FC<ChannelItem> = ({ snippet, id }) => {
  const { thumbnails, channelTitle, description } = snippet;

  const imgUrl =
    thumbnails?.high?.url ||
    thumbnails?.medium?.url ||
    thumbnails?.default?.url;

  return (
    <VStack as='article' spacing={4} align={'stretch'} w='full'>
      <Center>
        <AspectRatio ratio={1} w='50%' rounded={'full'} bgColor='gray.100'>
          <Image
            src={imgUrl}
            loading='lazy'
            alt={channelTitle}
            rounded={'full'}
          />
        </AspectRatio>
      </Center>
      <VStack align={'stretch'} h='full'>
        <Heading as='h1' size='lg'>
          {channelTitle}
        </Heading>
        <Text noOfLines={3}>{description}</Text>
      </VStack>
      <Center>
        <NextChakraButton href={`/${id.channelId}`}>
          Goto Channel
        </NextChakraButton>
      </Center>
    </VStack>
  );
};
