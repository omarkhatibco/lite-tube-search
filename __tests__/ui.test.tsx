import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SearchForm } from 'components/forms';
import { ChannelListItem } from 'components/SearchPage';
import { response } from '__mocks__';

describe('SearchForm', () => {
  const onSubmit = jest.fn();
  const onReset = jest.fn();
  it('should Works', async () => {
    render(<SearchForm onSubmit={onSubmit} onReset={onReset} />);

    await userEvent.type(
      screen.getByPlaceholderText(/search channels.../i),
      'tech',
    );
    await userEvent.click(screen.getByTestId('submit'));

    await waitFor(() => {
      expect(onSubmit).toHaveBeenCalledWith('tech');
    });

    await userEvent.click(screen.getByTestId('clear-search'));

    await waitFor(() => {
      expect(onReset).toBeCalledTimes(1);
    });
  });
});

describe('ChannelListItem', () => {
  const { items } = response;
  const item = items[0];
  it('match SnapShot', async () => {
    const { container } = render(<ChannelListItem {...item} />);

    expect(container).toMatchSnapshot();
  });
});
