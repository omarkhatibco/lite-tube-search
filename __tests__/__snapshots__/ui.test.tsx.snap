// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelListItem match SnapShot 1`] = `
<div>
  <article
    class="chakra-stack css-1v0sbj7"
  >
    <div
      class="css-gmuwbf"
    >
      <div
        class="chakra-aspect-ratio css-1qgkkae"
      >
        <img
          alt="Linus Tech Tips"
          class="chakra-image css-15k6n5f"
          loading="lazy"
          src=""
        />
      </div>
    </div>
    <div
      class="chakra-stack css-fnkifr"
    >
      <h1
        class="chakra-heading css-0"
      >
        Linus Tech Tips
      </h1>
      <p
        class="chakra-text css-1cp6yu0"
      >
        Looking for a Tech YouTuber? Linus Tech Tips is a passionate team of "professionally curious" experts in consumer technology ...
      </p>
    </div>
    <div
      class="css-gmuwbf"
    >
      <a
        class="chakra-button css-4xx2wk"
        href="/UCXuqSBlHAE6Xw-yeJA0Tunw"
      >
        Goto Channel
      </a>
    </div>
  </article>
</div>
`;
